syntax = "proto2";

package enerlyzer;

message dataset {
    required double logger_time                 =  29;
    required float current_l1_avg              =  1;
    required float current_l2_avg              =  2;
    required float current_l3_avg              =  3;
    required float voltage_l21_avg             =  4;
    required float voltage_l32_avg             =  5;
    required float voltage_l13_avg             =  6;
    required float current_l1_eff              =  7;
    required float current_l2_eff              =  8;
    required float current_l3_eff              =  9;
    required float voltage_l21_eff             =  10;
    required float voltage_l32_eff             =  11;
    required float voltage_l13_eff             =  12;
    required float current_l1_max              =  13;
    required float current_l2_max              =  14;
    required float current_l3_max              =  15;
    required float voltage_l21_max             =  16;
    required float voltage_l32_max             =  17;
    required float voltage_l13_max             =  18;
    required float temperature_l1              =  19;
    required float temperature_l2              =  20;
    required float temperature_l3              =  21;
    required float voltage_aux                 =  22;
    required float frequency_Hz                =  23;
    required float power                       =  24;
    required float external_current_sensor     =  25;
    required float supply_voltage              =  26;
    required float cpu_temperature             =  27;
    required float coin_cell_mv                =  28;
    required float used_storage_percent        =  30;
    required double energy_start                = 31;
    required double energy_Wh                   = 32;
    required float raspberry_temp_c             = 33;
    required float raspberry_mem_free_percent   = 34;
    
    required bool raspberry_under_voltage               = 35;
    required bool raspberry_frequency_capped            = 36;
    required bool raspberry_currently_throttled         = 37;
    required bool raspberry_under_voltage_has_occurred  = 38;
    required bool raspberry_frequency_capped_has_occurred = 39;
    required bool raspberry_throttling_has_occurred     = 40;
    
    
}
